src/toil/lib/aws/iam.py

```diff
@@ -0,0 +1,182 @@
+
+import logging
+import boto3
+from toil.lib.aws import zone_to_region
+from toil.provisioners.aws import get_best_aws_zone
+from functools import lru_cache
+from typing import Any, List, Dict, Set, cast
+
+from mypy_boto3_iam import IAMClient
+from mypy_boto3_iam.type_defs import GetRolePolicyResponseTypeDef
+from toil.lib.aws.session import AWSConnectionManager
+
+
+logger = logging.getLogger(__name__)
+
+_CLUSTER_LAUNCHING_PERMISSIONS = {"iam:CreateRole",
+                                  "iam:CreateInstanceProfile",
+                                  "iam:TagInstanceProfile",
+                                  "iam:DeleteRole",
+                                  "iam:DeleteRoleProfile",
+                                  "iam:ListAttatchedRolePolicies",
+                                  "iam:ListPolicies",
+                                  "iam:ListRoleTags",
+                                  "iam:PutRolePolicy",
+                                  "iam:RemoveRoleFromInstanceProfile",
+                                  "iam:TagRole"
+                                  }
+
+
+def check_policy_warnings(allowed_actions: Dict[str, List[str]] = {'*': []}, launching_perms: Set[str] = _CLUSTER_LAUNCHING_PERMISSIONS) -> bool:
+    """
+    Check whether necessary permissions are permitted for AWS
+
+    :param allowed_actions: Dictionary containing actions allowed by resource
+    :param launching_perms: Set of required actions to launch a cluster on AWS
+    """
+    permissions = [x for x in launching_perms if check_permission_allowed(x, allowed_actions["*"])]
```

Comment:
Here's where we're getting the requirement that all the grants be on the `*` resource. We should probably note that in a comment, so that when we come back to change it later we know where to start.